### ISSUE TITLE
Add search filters to the query string to make it possible to link to search results

### ISSUE THREAD

**Issue by a-rodin** [timberio/vector#2139]
It would be nice to be able to share search results from pages like [the sinks reference](https://vector.dev/docs/reference/sinks/).

Today I wanted to share a link which would list all `aws_*` sinks in a single page, and one way to do it could have been using the search page with pre-filled query. For example, it might be possible to manipulate the URL using the history API to automatically set the current page URL look like `https://vector.dev/docs/reference/sinks/?search=aws_` when the user inputs search, and then parse these parameters when loading the corresponding page if they are present.

**Comment by binarylogic** [comment]
I agree! The way docusuarus is setup with React router makes it difficult to change the window location on the fly. I'll dig in more and see if there is an easy way. But I do have query params available:

```
https://vector.dev/components/?providers[]=AWS
```

**Comment by a-rodin** [comment]
I actually did something similar in #1962 for tab selectors using effects and event listeners which have `urlKey` property:

https://github.com/timberio/vector/blob/7d4d1b5b548a129cc8af9b467926cb78fc622adf/website/src/theme/Tabs/index.js#L51-L106

I think the code from here (the content of `useEffect` and the `onSelectedValue` function) can be reused in the search component. It makes the component not pure, but I think it is not an issue for the documentation website, as we can guarantee that the names of the parameters don't clash for different components.
